rnwinrt/rnwinrt/react/strings/base.cpp

```diff
@@ -205,22 +205,29 @@ jsi::Value static_enum_data::create(jsi::Runtime& runtime) const
     return jsi::Value(runtime, jsi::Object::createFromHostObject(runtime, std::make_shared<projected_enum>(this)));
 }
 
-jsi::Value static_enum_data::get_value(std::string_view valueName) const
+jsi::Value static_enum_data::get_value(jsi::Runtime& runtime, std::string_view valueName) const
 {
     // TODO: It would also be rather simple to ensure that the array is sorted and do a binary search, however the
     // number of enum elements is typically pretty small, so that may actually be harmful
-    auto itr = std::find_if(values.begin(), values.end(), [&](auto& mapping) { return mapping.name == valueName; });
+    auto itr = std::find_if(values.begin(), values.end(),
+        [&](auto& mapping) { return mapping.name == valueName || std::to_string((long)mapping.value) == valueName; });
```

Comment:
For the cast of `mapping.value`, consider using `static_cast<int64_t>(mapping.value)` instead (as that is a fixed-width int type, as opposed to a compiler-specific integral type)

Reply:
This should use `std::underlying_type_t` instead of assuming any one type. E.g. `static_cast<std::underlying_type_t<decltype(mapping.value)>(mapping.value)`

Reply:
Simply calling `std::to_string` on the value gives something like `"1.000000"` which is why I added the cast to `long` first. I also looked into storing the string value in the values list, but this meant having to use a non `constexpr` constructor, which had lots of ripple effects.

Reply:
> I also looked into storing the string value in the values list, but this meant having to use a non constexpr constructor

I'm not sure how you got to that conclusion. String literals don't violate `constexpr` rules (in fact, `mapping.name` is already a string literal)

Reply:
I'm referring to calling `std::to_string` in the constructor.

Reply:
> I'm referring to calling `std::to_string` in the constructor.

My suggestion was not to call `std::to_string` in the constructor. It was to include the string representation of the numeric value in the generated code (e.g. as a string literal)

Reply:
That said, I'm not entirely sure that's the correct approach here
1. It increases binary size
2. There's no guarantee that the string passed in matches our string representation. E.g. the user could pass `"1.0"`, `"1e0"`, `"0x01"`, etc. IMO this should do my suggestion #2 from below and try and convert the string to a numeric value and do the comparison

Reply:
Javascript does the numeric conversion. For example, the following all work with my above code:

```
console.log(TestComponent.TestEnum[1.0], TestComponent.TestEnum[1e0], TestComponent.TestEnum[0x01]);

Output:
first first first
```

On the other hand, the following should not work (it doesn't for native JS objects): `TestComponent.TestEnum["0x01"]`.

So, I don't think we should convert the string to double and then compare to the values.

Reply:
> the following should not work (it doesn't for native JS objects)

Interesting to know. I'm curious. Does `TestComponent.TestEnum["1"]` work and should it work? I feel like the failure comes from the fact that you're using a string as opposed to a number

Reply:
Perhaps the solution is to keep the argument as `jsi::PropNameID` and do the lookup depending on whether it holds a number vs a string

Reply:
> Interesting to know. I'm curious. Does `TestComponent.TestEnum["1"]` work and should it work? I feel like the failure comes from the fact that you're using a string as opposed to a number

Yes, `TestComponent.TestEnum["1"]` also works.

Reply:
Huh, I see. I wonder if there's something in the JS standard that controls which strings are allowed to be treated as numbers/indices here. Ideally, we would use/replicate that logic. IMO short of trying to do that, it's not really worth it to try and get exact 100% parity (we already don't in other places). If we allow something like `Enum["1.0"]` to work when it ideally shouldn't, that's not the end of the world

Reply:
> For the cast of `mapping.value`, consider using `static_cast<int32_t>(mapping.value)` instead (as that is a fixed-width int type, as opposed to a compiler-specific integral type).

I've made this change.